apps/payroll/contracts/Payroll.sol

```diff
@@ -0,0 +1,586 @@
+pragma solidity 0.4.15;
+
+import "./oracle/OracleInterface.sol";
+
+import "@aragon/core/contracts/apps/App.sol";
+import "@aragon/core/contracts/common/Initializable.sol";
+import "@aragon/core/contracts/common/EtherToken.sol";
+import "@aragon/core/contracts/common/erc677/ERC677Receiver.sol";
+
+import "@aragon/core/contracts/zeppelin/token/ERC20.sol";
+import "@aragon/core/contracts/zeppelin/math/SafeMath.sol";
+
+import "@aragon/apps-finance/contracts/Finance.sol";
+//import "@aragon/apps-vault/contracts/Vault.sol";
+
+
+/**
+ * @title Payroll in multiple currencies
+ * @notice For the sake of simplicity lets asume USD is a ERC20 token
+ * Also lets asume we can 100% trust the exchange rate oracle
+ */
+contract Payroll is App, Initializable, ERC677Receiver {
+    using SafeMath for uint;
+
+    bytes32 constant public OWNER_WRITE_ROLE = bytes32(1);
+    bytes32 constant public OWNER_READ_ROLE = bytes32(2);
+    bytes32 constant public EMPLOYEE_ROLE = bytes32(3);
```

Comment:
In order to make the employee role possible to be used, a [group](https://github.com/aragon/aragon-apps/tree/master/apps/group) with all employees would need to be maintained in parallel, which isn't too efficient. In this case I would replace having kernel auth for employees, and would do a local modifier in this contract to check if the sender is an employee.

Reply:
I agree.